composer/algorithms/swa/swa.py

```diff
@@ -41,43 +42,79 @@ class SWA(Algorithm):
             trainer = Trainer(
                 model=model,
                 train_dataloader=train_dataloader,
                 eval_dataloader=eval_dataloader,
-                max_duration="1ep",
+                max_duration="10ep",
                 algorithms=[swa_algorithm],
                 optimizers=[optimizer]
             )
 
     Args:
-        swa_start (float): Fraction of training completed before stochastic weight
-            averaging is applied. Defalt = ``0.8``.
+        swa_start (float, optional): Fraction of training completed before stochastic
+            weight averaging begins. Defalt = ``0.7``.
+        swa_end (float, optional): Fraction of training completed before the baseline
+            (non-averaged) model is replaced with the stochastic weight averaged model.
+            It's important to have at least one epoch of training after the baseline model
+            is replaced by the SWA model so that the SWA model can have its buffers (most
+            importantly its batch norm statistics) updated. If ``swa_end`` occurs during
+            the final epoch of training (e.g. ``swa_end = 0.9`` and ``max_duration =
+            "5ep"``), the SWA model will not have its buffers updated, which can
+            negatively impact accuracy, so ensure ``swa_end`` <
+            :math:`\\frac{N_{epochs}-1}{N_{epochs}}`. Default = ``0.97``.
+        schedule_swa_lr (bool, optional): Flag to determine whether to apply an
+            SWA-specific LR schedule during the period in which SWA is active. Default =
+            ``False``.
+        anneal_strategy (str, optional): SWA learning rate annealing schedule strategy.
+            "linear" for linear annealing, "cos" for cosine annealing. Default =
+            ``"linear"``.
         anneal_epochs (int, optional): Number of epochs over which to anneal SWA
             learning rate. Default = ``10``.
-        swa_lr (float, optional): The final learning rate used for weight averaging
+        swa_lr (float, optional): The final learning rate to anneal towards with SWA LR
+            scheduler. Set to ``None`` for no annealing.
     """
 
-    def __init__(self, swa_start: float = 0.8, anneal_epochs: int = 10, swa_lr: Optional[float] = None):
+    def __init__(self,
+                 swa_start: float = 0.7,
+                 swa_end: float = 0.97,
+                 schedule_swa_lr: bool = False,
+                 anneal_strategy: str = "linear",
+                 anneal_epochs: int = 10,
+                 swa_lr: Optional[float] = None):
         self.swa_start = swa_start
+        self.swa_end = swa_end
+        self.schedule_swa_lr = schedule_swa_lr
+        self.anneal_strategy = anneal_strategy
         self.anneal_epochs = anneal_epochs
         self.swa_lr = swa_lr
         self.swa_model: Optional[torch.nn.Module] = None
+        self.swa_completed = False
+
+        # Check annealing_strategy string
+        if self.anneal_strategy.lower() in ["linear", "lin"]:
```

Comment:
perhaps just support `linear` and `cosine` , and not the shorthand abbreviations?

Reply:
`SWALR` (the SWA LR scheduler) accepts `cos` or `linear`. We could have it match PT exactly, but I think it's weird to strictly enforce one being abbreviated and the other not.

Reply:
ah ok sounds good.